language/diem-framework/modules/DesignatedDealer.move

```diff
@@ -158,138 +115,41 @@ module DesignatedDealer {
-    }
-
-
-    public fun update_tier<CoinType: store>(
-        tc_account: &signer,
-        dd_addr: address,
-        tier_index: u64,
-        new_upperbound: u64
-    ) acquires TierInfo {
-        Roles::assert_treasury_compliance(tc_account);
-        assert(exists<TierInfo<CoinType>>(dd_addr), Errors::not_published(EDEALER));
-        let tiers = &mut borrow_global_mut<TierInfo<CoinType>>(dd_addr).tiers;
-        let number_of_tiers = Vector::length(tiers);
-        assert(tier_index < number_of_tiers, Errors::invalid_argument(EINVALID_TIER_INDEX));
-        // Make sure that this new start for the tier is consistent with the tier above and below it.
-        let tier = Vector::borrow(tiers, tier_index);
-        if (*tier == new_upperbound) return;
-        if (*tier < new_upperbound) {
-            let next_tier_index = tier_index + 1;
-            if (next_tier_index < number_of_tiers) {
-                assert(
-                    new_upperbound < *Vector::borrow(tiers, next_tier_index),
-                    Errors::invalid_argument(EINVALID_TIER_START)
-                );
-            };
-        };
-        if (*tier > new_upperbound && tier_index > 0) {
-            let prev_tier_index = tier_index - 1;
-            assert(
-                new_upperbound > *Vector::borrow(tiers, prev_tier_index),
-                Errors::invalid_argument(EINVALID_TIER_START)
-            );
-        };
-        *Vector::borrow_mut(tiers, tier_index) = new_upperbound;
-    }
-    spec fun update_tier {
-        pragma opaque;
-
-        include Roles::AbortsIfNotTreasuryCompliance{account: tc_account};
-        include AbortsIfNoTierInfo<CoinType>;
-        let tier_info = global<TierInfo<CoinType>>(dd_addr);
-        aborts_if tier_index >= len(tier_info.tiers) with Errors::INVALID_ARGUMENT;
-        aborts_if tier_index > 0 && tier_info.tiers[tier_index - 1] >= new_upperbound with Errors::INVALID_ARGUMENT;
-        aborts_if tier_index + 1 < len(tier_info.tiers) && tier_info.tiers[tier_index + 1] <= new_upperbound with Errors::INVALID_ARGUMENT;
-
-        modifies global<TierInfo<CoinType>>(dd_addr);
-        ensures exists<TierInfo<CoinType>>(dd_addr);
-        ensures global<TierInfo<CoinType>>(dd_addr) ==
-            TierInfo<CoinType> {
-                window_start: old(global<TierInfo<CoinType>>(dd_addr)).window_start,
-                window_inflow: old(global<TierInfo<CoinType>>(dd_addr)).window_inflow,
-                tiers: update_vector(old(global<TierInfo<CoinType>>(dd_addr)).tiers, tier_index, new_upperbound),
-            };
-    }
-
     public fun tiered_mint<CoinType: store>(
         tc_account: &signer,
         amount: u64,
         dd_addr: address,
-        tier_index: u64,
+        // tiers are deprecated. We continue to accept this argument for backward
+        // compatibility, but it will be ignored.
+        _tier_index: u64,
     ): Diem::Diem<CoinType> acquires Dealer, TierInfo {
```

Comment:
Just checking we do indeed want to leave `TierInfo` here even after the first call that resource won't exist (but we need it for the first call)?

Reply:
Yes, the bytecode verifier forces us to have `acquires TierInfo` anywhere that there's a `move_from<TierInfo>`.